SingularityService/src/test/java/com/hubspot/singularity/data/InactiveSlaveManagerTest.java

```diff
@@ -0,0 +1,27 @@
+package com.hubspot.singularity.data;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.google.inject.Inject;
+import com.hubspot.singularity.SingularityTestBaseNoDb;
+
+public class InactiveSlaveManagerTest extends SingularityTestBaseNoDb {
+  @Inject
+  InactiveSlaveManager inactiveSlaveManager;
+
+  @Test
+  public void itShouldContainAnInactiveHostWhenHostDeactivated() {
+    inactiveSlaveManager.deactiveSlave("localhost");
+
+    Assert.assertTrue(inactiveSlaveManager.getInactiveSlaves().contains("localhost"));
+  }
+
+  @Test
+  public void itShouldNotContainHostAfterActivatingHost() {
```

Comment:
to extend these tests, you can have the class extend `SingularitySchedulerTestBase`. In there is a `resourceOffers` method which mocks the sending of an offer, allowing you to trigger the code path of getting an offer from a new slave.